### PR TITLE
MM-22616 - Add marketplace plugin docs

### PR DESCRIPTION
**Pull request by teyotan**
#### Summary
Add marketplace plugin API documentation.
`page`, `per_page`, and `server_version` params on GET method is not yet implemented, please kindly check.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/13899

### CODE REVIEW

File: v4/source/definitions.yaml [hunk 1]
```diff
@@ -2236,6 +2236,36 @@ definitions:
         type: object
         description: Settings schema used to define the System Console UI for the plugin.
 
+  MarketplacePlugin:
+    type: object
+    properties:
+      homepage_url:
+        type: string
+        description: "URL that leads to the homepage ot the plugin."
```
Comment (lieut-data) [inline v4/source/definitions.yaml]:
```suggestion
        description: "URL that leads to the homepage of the plugin."
```

File: v4/source/definitions.yaml [hunk 2]
```diff
@@ -2236,6 +2236,36 @@ definitions:
...
+      icon_data:
+        type: string
+        description: "Base64 that represents icon of the plugin."
```
Comment (lieut-data) [inline v4/source/definitions.yaml]:
```suggestion
        description: "Base64 encoding of a plugin icon SVG."
```

File: v4/source/definitions.yaml [hunk 3]
```diff
@@ -2236,6 +2236,36 @@ definitions:
...
+        type: string
+        description: URL that leads to the release notes of the plugin.
+      labels:
+        type: array
+        items:
+          type: string
+        description: A list of the plugin labels.
+      signature:
+        type: string
+        description: "Signature of the plugin"
```
Comment (lieut-data) [inline v4/source/definitions.yaml]: Any particular reason for this to be in quotes?
```suggestion
        description: Signature of the plugin
```

File: v4/source/definitions.yaml [hunk 4]
```diff
@@ -2236,6 +2236,36 @@ definitions:
...
+      manifest:
+        description: Manifest of the plugin
+        $ref: '#/definitions/PluginManifest'
+      installed_version:
+        type: string
+        description: Version number of the plugin.
```
Comment (lieut-data) [inline v4/source/definitions.yaml]:
```suggestion
        description: Version number of the already installed plugin, if any.
```

File: v4/source/plugins.yaml [hunk 5]
```diff
@@ -323,3 +323,131 @@
...
+            required:
+              - id
+              - version
+            properties:
+              id:
+                type: string
+                description: The plugin ID to install.
+              version:
+                type: string
+                description: The version of plugin to install.
```
Comment (lieut-data) [inline v4/source/plugins.yaml]:
```suggestion
                description: The version of the plugin to install.
```

File: v4/source/plugins.yaml [hunk 6]
```diff
@@ -323,3 +323,131 @@
...
+          required: true
+          schema:
+            type: object
+            required:
+              - id
+              - version
+            properties:
+              id:
+                type: string
+                description: The plugin ID to install.
```
Comment (lieut-data) [inline v4/source/plugins.yaml]:
```suggestion
                description: The ID of the plugin to install.
```

File: v4/source/plugins.yaml [hunk 7]
```diff
@@ -323,3 +323,131 @@
...
+        Installs a plugin listed in the marketplace server.
+
+        ##### Permissions
+        Must have `manage_system` permission.
+
+        __Minimum server version__: 5.16
+      parameters:
+        - name: body
+          in: body
+          description: Plugin data to install.
```
Comment (lieut-data) [inline v4/source/plugins.yaml]:
```
          description: The metadata identifying the plugin to install.
```

File: v4/source/plugins.yaml [hunk 8]
```diff
@@ -323,3 +323,131 @@
...
+              Version: "0.1.2",
+            }
+
+            ok, resp = Client.InstallMarketplacePlugin(plugin)
+    get:
+      tags:
+        - plugins
+      summary: Gets all the marketplace plugins
+      description: |
+        Gets all plugins from the marketplace server and plugins that are installed locally
```
Comment (lieut-data) [inline v4/source/plugins.yaml]:
```suggestion
        Gets all plugins from the marketplace server, merging data from locally installed plugins as well as prepackaged plugins shipped with the server.
```

File: v4/source/plugins.yaml [hunk 9]
```diff
@@ -323,3 +323,131 @@
...
+          required: false
+          type: string
+        - name: server_version
+          in: query
+          description: Set to filter minimum plugin server version. (not yet implemented)
+          required: false
+          type: string
+        - name: local_only
+          in: query
+          description: Set true to only retrive local plugins only
```
Comment (lieut-data) [inline v4/source/plugins.yaml]:
```suggestion
          description: Set true to only retrieve local plugins.
```

File: v4/source/plugins.yaml [hunk 10]
```diff
@@ -323,3 +323,131 @@
...
+      x-code-samples:
+        - lang: 'Go'
+          source: |
+            import "github.com/mattermost/mattermost-server/model"
+
+            Client := model.NewAPIv4Client("https://your-mattermost-url.com")
+            Client.Login("email@domain.com", "Password1")
+
+            filter := &model.MarketplacePluginFilter{
+              Page: "antivirus",
```
Comment (lieut-data) [inline v4/source/plugins.yaml]: The examples here don't seem right -- can you clarify?

**Comment by hanzei** [comment]
Hey @ali-farooq0, gentle reminder to review this PR

File: v4/source/definitions.yaml [hunk 11]
```diff
@@ -2236,6 +2236,36 @@ definitions:
...
+        type: string
+        description: URL that leads to the release notes of the plugin.
+      labels:
+        type: array
+        items:
+          type: string
+        description: A list of the plugin labels.
+      signature:
+        type: string
+        description: Signature of the plugin
```
Comment (ali-farooq0) [inline v4/source/definitions.yaml]:
```suggestion
        description: Base64 encoded signature of the plugin.
```

File: v4/source/definitions.yaml [hunk 12]
```diff
@@ -2236,6 +2236,36 @@ definitions:
...
+      manifest:
+        description: Manifest of the plugin
```
Comment (ali-farooq0) [inline v4/source/definitions.yaml]:
```suggestion
        description: Manifest of the plugin.
```

**Comment by metanerd** [comment]
@teyotan thank you for the PR. I will merge this, please ignore the circleci hiccup reporting back the wrong CI status.

**Pull request merged**